File: 03-init/src/main.c
#include <stdio.h>
#include <stdbool.h>

typedef struct {
    int uid;
    int gid;
} user_t;

void init_user(user_t *user, int uid) {
    int gid;

    if (uid % 2 == 0) {
	gid = uid;
    } else {
	gid = 1;
    }
    
    user->uid = uid;
    user->gid = gid;
}

void print_user(user_t *user) {
    printf("User, uid: %i, gid: %i\n", user->uid, user->gid);
}

int main(void) {    
    user_t user;

    print_user(&user);
    init_user(&user, 42);
    print_user(&user);
}

File: 03-init/src/main.rs
struct User {
    uid: i32,
    gid: i32,
}

impl User {
    fn new(uid: i32) -> Self {
        let gid = if uid % 2 == 0 { uid } else { 1 };
        Self { uid, gid }
    }
}

fn print_user(user: &User) {
    println!("User, id: {}, gid: {}", user.uid, user.gid);
}

fn main() {
    let user = User::new(42);

    print_user(&user);
}
